parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```
@@ -248,6 +284,351 @@ private CompressionCodecName codec() {
+
+        int pageSize = Integer.parseInt(config.getOrDefault(
+            PARQUET_PAGE_SIZE_BYTES, PARQUET_PAGE_SIZE_BYTES_DEFAULT));
+
+        int dictionaryPageSize = Integer.parseInt(config.getOrDefault(
+            PARQUET_DICT_SIZE_BYTES, PARQUET_DICT_SIZE_BYTES_DEFAULT));
+
+        String codecAsString = config.getOrDefault(PARQUET_COMPRESSION, PARQUET_COMPRESSION_DEFAULT);
+        CompressionCodecName codec = toCodec(codecAsString);
+
+        String compressionLevel = config.getOrDefault(PARQUET_COMPRESSION_LEVEL, PARQUET_COMPRESSION_LEVEL_DEFAULT);
+
+        return new Context(rowGroupSize, pageSize, dictionaryPageSize, codec, compressionLevel);
+      }
+
+      static Context deleteContext(Map<String, String> config) {
+        // default delete config using data config
+        Context dataContext = dataContext(config);
+
+        int rowGroupSize = PropertyUtil.propertyAsInt(config,
+            DELETE_PARQUET_ROW_GROUP_SIZE_BYTES, dataContext.rowGroupSize());
+
+        int pageSize = PropertyUtil.propertyAsInt(config,
+            DELETE_PARQUET_PAGE_SIZE_BYTES, dataContext.pageSize());
+
+        int dictionaryPageSize = PropertyUtil.propertyAsInt(config,
+            DELETE_PARQUET_DICT_SIZE_BYTES, dataContext.dictionaryPageSize());
+
+        String codecAsString = config.get(DELETE_PARQUET_COMPRESSION);
+        CompressionCodecName codec = codecAsString != null ? toCodec(codecAsString) : dataContext.codec();
+
+        String compressionLevel = config.getOrDefault(DELETE_PARQUET_COMPRESSION_LEVEL, dataContext.compressionLevel());
+
+        return new Context(rowGroupSize, pageSize, dictionaryPageSize, codec, compressionLevel);
+      }
+
+      private static CompressionCodecName toCodec(String codecAsString) {
+        try {
+          return CompressionCodecName.valueOf(codecAsString.toUpperCase(Locale.ENGLISH));
+        } catch (IllegalArgumentException e) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)